drivers/bms_ic/bq769x0/bq769x0.c

```diff
@@ -120,94 +120,67 @@ int bq769x0_write_byte(const struct device *dev, uint8_t reg_addr, uint8_t data)
     }
 }
 
-int bq769x0_read_byte(const struct device *dev, uint8_t reg_addr, uint8_t *byte)
+static int bq769x0_read_bytes(const struct device *dev, uint8_t reg_addr, uint8_t *data,
+                              size_t num_bytes)
 {
     const struct bms_ic_bq769x0_config *dev_config = dev->config;
     const struct bms_ic_bq769x0_data *dev_data = dev->data;
-    uint8_t buf[3];
+    uint8_t buf[5] = {
+        (dev_config->i2c.addr << 1) | 1U, /* target address for CRC calculation */
+    };
     int err;
 
-    err = i2c_write_dt(&dev_config->i2c, &reg_addr, 1);
-    if (err != 0) {
-        return err;
+    if (num_bytes < 1 || num_bytes > 2) {
+        return -EINVAL;
     }
 
     if (dev_data->crc_enabled) {
-        /* CRC is calculated over the slave address (incl. R/W bit) and data */
-        buf[0] = (dev_config->i2c.addr << 1) | 1U;
-        int attempts = 1;
-        while (true) {
-            err = i2c_read_dt(&dev_config->i2c, buf + 1, 2);
+        for (int attempts = 1; attempts <= BQ769X0_READ_MAX_ATTEMPTS; attempts++) {
+            err = i2c_write_read_dt(&dev_config->i2c, &reg_addr, 1, buf + 1, num_bytes);
```

Comment:
you need to read double the num_bytes ( data crc data crc )

Reply:
Good spot, thanks! Fixed.